src/kernels/installer/productInstaller.ts

```diff
@@ -117,6 +117,9 @@ abstract class BaseInstaller {
         if (!installer) {
             return InstallerResponse.Ignore;
         }
+        if (cancelTokenSource.token.isCancellationRequested) {
```

Comment:
You'll see these in a few places, while running the tests found that these were causing issues.
Basically we had stopped cell execution (either closing a document or interrupt), & these code paths would continue running.